experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,248 @@
+simulation start and end times.
+
+The default periods are:
+- 1 month for simulations longer than 90 days
+- 10 days for simulations longer than 30 days
+- 1 day for simulations longer than 1 day
+- 1 hour for simulations shorter than 1 day
+
+# Arguments
+- `t_start`: The start time of the simulation
+- `t_end`: The end time of the simulation
+
+# Returns
+- `period`: A String of how often to average and output diagnostics
+- `calendar_dt`: A DateTime interval representing the period
+"""
+function get_diag_period(t_start, t_end)
+    sim_duration = t_end - t_start
+    secs_per_day = 86400
+    if sim_duration >= 90 * secs_per_day
+        # if duration >= 90 days, take monthly means
+        period = "1months"
+        calendar_dt = Dates.Month(1)
+    elseif sim_duration >= 30 * secs_per_day
+        # if duration >= 30 days, take means over 10 days
+        period = "10days"
+        calendar_dt = Dates.Day(10)
+    elseif sim_duration >= secs_per_day
+        # if duration >= 1 day, take daily means
+        period = "1days"
+        calendar_dt = Dates.Day(1)
+    else
+        # if duration < 1 day, take hourly means
+        period = "1hours"
+        calendar_dt = Dates.Hour(1)
+    end
+    return (period, calendar_dt)
+end
+
+"""
+    parse_component_dts!(config_dict)
+
+Check which timesteps are specified in the config file, and use them to choose
+the correct timestep for each component model.
+If all component timesteps `dt_\$component` are specified in the config file, use those
+and remove `dt` if it was provided.
+Otherwise, use the generic component timestep `dt` specified in the config file.
+If some (but not all) component timesteps and the generic timestep `dt` are specified,
+use the generic timestep and remove the others from the config dict.
+
+# Arguments
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+
+# Returns
+- `component_dt_dict`: A dictionary mapping component model names to their respective timesteps
+"""
+function parse_component_dts!(config_dict)
+    # Retrieve coupling timestep
+    Δt_cpl = Float64(Utilities.time_to_seconds(config_dict["dt_cpl"]))
+
+    # Specify component model names
+    component_dt_names = ["dt_atmos", "dt_land", "dt_ocean", "dt_seaice"]
+    component_dt_dict = Dict{String, Float64}()
+    # check if all component dt's are specified
+    if all(key -> !isnothing(config_dict[key]), component_dt_names)
+        # when all component dt's are specified, ignore the dt field
+        if haskey(config_dict, "dt")
+            @warn "Removing dt in favor of individual component dt's"
+            delete!(config_dict, "dt")
+        end
+        for key in component_dt_names
+            component_dt = Float64(Utilities.time_to_seconds(config_dict[key]))
+            @assert Δt_cpl % component_dt == 0.0 "Coupler dt must be divisible by all component dt's\n dt_cpl = $Δt_cpl\n $key = $component_dt"
```

Comment:
It's typically not a good idea to do exact comparison involving floating points. For example, 1e-300 != 0.0 so this would fail